### DISCUSSION TITLE
Add Support for Auto-Fill Addresses in Block Checkout System

### DISCUSSION THREAD

**Discussion by archiveofwork** [woocommerce/woocommerce#47834] [Checkout Flow]
As a WooCommerce user, I am requesting the addition of an auto-fill address feature in the new block checkout system. This feature is increasingly becoming a standard in e-commerce platforms and is essential for enhancing user experience and reducing cart abandonment rates.

- User Experience: Auto-fill address functionality significantly improves the user experience by reducing the time and effort required to enter address details manually.
- Industry Standard: Major competitors like Shopify already offer this feature, setting a higher standard for customer expectations.
- Error Reduction: Automatically populated addresses reduce the likelihood of typographical errors, ensuring more accurate shipping information.
- Efficiency: Streamlining the checkout process can lead to higher conversion rates and lower cart abandonment.

**Implementation**
To implement this feature, integrating a reliable API such as the Google Places API for address auto-completion would be highly beneficial. This API provides comprehensive and accurate address data, which can be used to auto-populate address fields as the user types.
Example Workflow

**Example Workflow**
- User Begins Typing: As the user starts typing their address in the checkout form, a dropdown list of suggested addresses appears.
- User Selects Address: The user selects the correct address from the suggestions.
- Fields Auto-Populate: The selected address is used to auto-fill the relevant fields in the checkout form.

**Benefits**
- Improved User Satisfaction: Customers enjoy a faster, easier checkout process.
- Increased Sales: Reducing friction in the checkout process can lead to higher sales conversion rates.
- Competitive Advantage: Offering this feature aligns WooCommerce with industry leaders and meets customer expectations.

Thank you for considering this feature request. I look forward to seeing this enhancement in future updates.

**Comment by senadir** [comment]
Thank you for the feedback, this is something that's on our radar and we had POC before, we will prioritize this again soon, one main thing we need to figure out is better key management, we ideally don't want merchants to fiddle out with keys and looking at some sort of host level support here + ability to set your own key.
